schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -368,6 +351,25 @@ public SchemaMetadataInfo getSchemaMetadata(String schemaName) {
         return schemaMetadataStorable != null ? schemaMetadataStorable.toSchemaMetadataInfo() : null;
     }
 
+
+    @Override
+    public SchemaMetadataInfo updateSchemaMetadata(String schemaName, SchemaMetadata schemaMetadata) {
+        if (!schemaName.equals(schemaMetadata.getName())) {
+            throw new IllegalArgumentException("schemaName must match the name in schemaMetadata");
+        }
+        SchemaMetadataStorable givenSchemaMetadataStorable = new SchemaMetadataStorable();
+        givenSchemaMetadataStorable.setName(schemaName);
+
+        SchemaMetadataStorable schemaMetadataStorable = storageManager.get(givenSchemaMetadataStorable.getStorableKey());
+        if (schemaMetadataStorable != null) {
+            schemaMetadataStorable = SchemaMetadataStorable.updateSchemaMetadata(schemaMetadataStorable, schemaMetadata);
```

Comment:
@michaelandrepearce I have already mentioned [here](https://github.com/hortonworks/registry/pull/175#pullrequestreview-48816224) that this can be merged in after resolving conflicts. I would like to have lifecycle management of schema/versions #177 address the scenarios you have mentioned. Initial design will be shared in next 3-4 days, community can discuss to enhance that to address different usecases.